pygeoapi/provider/mvt_postgres.py

```diff
@@ -0,0 +1,263 @@
+        Initialize object
+
+        :param provider_def: provider definition
+
+        :returns: pygeoapi.provider.MVT.MVTPostgresProvider
+        """
+
+        super().__init__(provider_def)
+        
+        pg_def = deepcopy(provider_def)
+        del pg_def["options"]["zoom"]
+        self.postgres = PostgreSQLProvider(pg_def)
+        self.layer_name = provider_def["table"]
+        self.table = provider_def['table']
+        self.id_field = provider_def['id_field']
+        self.geom = provider_def.get('geom_field', 'geom')
+
+        LOGGER.debug(f'DB connection: {repr(self.postgres._engine.url)}')
+
+    def __repr__(self):
+        return f'<MVTPostgresProvider> {self.data}'
+
+    @property
+    def service_url(self):
+        return self._service_url
+
+    @property
+    def service_metadata_url(self):
+        return self._service_metadata_url
+
+    def get_layer(self):
+        """
+        Extracts layer name from url
+
+        :returns: layer name
+        """
+
+        return self.layer_name
+
+    def get_tiling_schemes(self):
+
+        return [
+                TileMatrixSetEnum.WEBMERCATORQUAD.value,
+                TileMatrixSetEnum.WORLDCRS84QUAD.value
+            ]
+
+    def get_tiles_service(self, baseurl=None, servicepath=None,
+                          dirpath=None, tile_type=None):
+        """
+        Gets mvt service description
+
+        :param baseurl: base URL of endpoint
+        :param servicepath: base path of URL
+        :param dirpath: directory basepath (equivalent of URL)
+        :param tile_type: tile format type
+
+        :returns: `dict` of item tile service
+        """
+
+        super().get_tiles_service(baseurl, servicepath,
+                                  dirpath, tile_type)
+
+        self._service_url = servicepath
+        return self.get_tms_links()
+
+    def get_tiles(self, layer=None, tileset=None,
+                  z=None, y=None, x=None, format_=None):
+        """
+        Gets tile
+
+        :param layer: mvt tile layer
+        :param tileset: mvt tileset
+        :param z: z index
+        :param y: y index
+        :param x: x index
+        :param format_: tile format
+
+        :returns: an encoded mvt tile
+        """
+        if format_ == 'mvt':
+            format_ = self.format_type
+
+        fields_arr = self.postgres.get_fields().keys()
+        fields = ', '.join(['"' + f + '"' for f in fields_arr])
+        if len(fields) != 0:
+            fields = ',' + fields
+        
+        query = ''
+        if tileset == TileMatrixSetEnum.WEBMERCATORQUAD.value.tileMatrixSet:
+            query = text("""
+                WITH                                                                  
+                    bounds AS (
+                        SELECT ST_TileEnvelope(:z, :y, :x) AS boundgeom
+                    ),
+                    mvtgeom AS (
+                        SELECT ST_AsMVTGeom(ST_Transform({geom}, 3857), bounds.boundgeom) AS geom {fields}
```

Comment:
I think converting curves to lines (ST_CurveToLine), could speed up queries in the case of more complex geometries.

Reply:
OK, I will try this out, [it looks like Martin also uses it](https://maplibre.org/martin/sources-pg-functions.html#simple-function). 

So by using `ST_CurveToLine`, will this provider also be able to support `CIRCULARSTRING`-like data types (that is what I understood from reading the [documentation](https://postgis.net/docs/ST_CurveToLine.html))? Or is it just going to be used for query performance?

Reply:
I think simplification, in general, yields performance; and in the case of tiles, we do not need the geometries to be so detailed.